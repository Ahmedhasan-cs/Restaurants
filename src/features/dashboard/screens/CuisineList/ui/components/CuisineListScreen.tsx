import {styles} from '../styles/cuisineListStyle';
import {View, FlatList, Alert} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import CuisineItem from './CuisineItem';
import {
  CuisinesNavigationProps,
  RouteNames,
} from '@navigators/dashboard/DashboardNavigationProps';
import {useCuisines} from '@features/dashboard/screens/CuisineList/hooks/useCuisines';
import {Cuisine} from '@util/api/models/CuisineResponse';
import {useEffect} from 'react';

const CuisineListScreen = ({route, navigation}: CuisinesNavigationProps) => {
  const {error, cuisineCounts, handleSelectedCuisine} = useCuisines();

  const handlePress = (cuisine: Cuisine): void => {
    console.log('Cuisine pressed', cuisine);
    handleSelectedCuisine(cuisine);
    navigation.navigate(RouteNames.Restaurants);
  };

  useEffect(() => {
    if (error) {
      Alert.alert('Error', error);
    }
  }, [error]);

  return (
    <SafeAreaView style={styles.flexOne}>
      <View style={styles.container}>
        <FlatList
          data={cuisineCounts}
          keyExtractor={item => item.title}
          renderItem={({item}) => (
            <CuisineItem cuisine={item} handlePress={handlePress} />
          )}
          contentContainerStyle={styles.list}
        />
      </View>
    </SafeAreaView>
  );
};

export default CuisineListScreen;
