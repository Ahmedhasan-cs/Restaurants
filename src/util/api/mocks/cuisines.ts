import { Cuisines } from "../models/CuisineResponse";

export const cuisinesData: Cuisines = {
    "chinese": {
      "open": [
        {
          "id": "6109200ecdbea9077786439c",
          "restaurantName": "Lunchpod",
          "shortDesc": "Relax your belly",
          "currency": "EUR",
          "deliveryCost": 2.15,
          "rating": 6,
          "minOrder": 10,
          "deliveryTime": "20-25 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_feedback002.jpg"
        },
        {
          "id": "6109200e7363c5fce8c4262e",
          "restaurantName": "Magneato",
          "shortDesc": "Soul for food",
          "currency": "EUR",
          "deliveryCost": 1.91,
          "rating": 6,
          "minOrder": 15,
          "deliveryTime": "15-20 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/umami_top10berlin003.jpg"
        },
        {
          "id": "6109200ead8772931ada1feb",
          "restaurantName": "Pheast",
          "shortDesc": "Paradise",
          "currency": "GBP",
          "deliveryCost": 0.12,
          "rating": 10,
          "minOrder": 14,
          "deliveryTime": "30-40 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/kimchi_princess_-_photo-ken_buslay-082.jpg"
        },
        {
          "id": "6109200e0800c681de35f1c2",
          "restaurantName": "Fuelton",
          "shortDesc": "A lovely place to eat",
          "currency": "EUR",
          "deliveryCost": 2.33,
          "rating": 8,
          "minOrder": 13,
          "deliveryTime": "30-40 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_asiatische-restaurants_foto_sticksnsushi_seabass_carpacio_1000x500.jpg"
        },
        {
          "id": "6109200e12f05a2b9b730c36",
          "restaurantName": "Xylar",
          "shortDesc": "Huger Heaven",
          "currency": "EUR",
          "deliveryCost": 1.26,
          "rating": 10,
          "minOrder": 13,
          "deliveryTime": "30-40 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/restaurant-kochu-karu_800x400_5.jpg"
        },
        {
          "id": "6109200eaf8b10675af3cc3a",
          "restaurantName": "Rotodyne",
          "shortDesc": "Huger Heaven",
          "currency": "EUR",
          "deliveryCost": 1.92,
          "rating": 8,
          "minOrder": 11,
          "deliveryTime": "20-25 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/panama-3411_c_philipp_langenheim_corina_schadendorf_inneneinrichtung_top10berlin_800x600px.jpg"
        },
        {
          "id": "6109200eb000b553af427a41",
          "restaurantName": "Zboo",
          "shortDesc": " Food for fun",
          "currency": "EUR",
          "deliveryCost": 2.53,
          "rating": 10,
          "minOrder": 10,
          "deliveryTime": "30-40 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/restaurant-kochu-karu_800x400_5.jpg"
        },
        {
          "id": "6109200ef1f2f8a0db99197f",
          "restaurantName": "Evidends",
          "shortDesc": "Soul for food",
          "currency": "EUR",
          "deliveryCost": 1.39,
          "rating": 8,
          "minOrder": 14,
          "deliveryTime": "30-40 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_good-friends.jpg"
        },
        {
          "id": "6109200e1d9c321c44e5b757",
          "restaurantName": "Rooforia",
          "shortDesc": "Soul for food",
          "currency": "EUR",
          "deliveryCost": 1.48,
          "rating": 7,
          "minOrder": 13,
          "deliveryTime": "15-20 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/restaurant-zen-berlin_asian-dining_sushi_800x400_1.jpg"
        }
      ],
      "close": [
        {
          "id": "6109200e07f1557c9dae06ff",
          "restaurantName": "Vicon",
          "shortDesc": "Paradise",
          "currency": "USD",
          "deliveryCost": 2.33,
          "rating": 10,
          "minOrder": 12,
          "deliveryTime": "30-40 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/nu-restaurant_berlin-charlottenburg_800x500_1.jpg"
        },
        {
          "id": "6109200ee93c6aec044a85fc",
          "restaurantName": "Satiance",
          "shortDesc": "Huger Heaven",
          "currency": "USD",
          "deliveryCost": 0.2,
          "rating": 8,
          "minOrder": 12,
          "deliveryTime": "15-20 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/bastard_fruehstueck_top10_berlin.jpg"
        },
        {
          "id": "6109200e95e3be726fe9e8e1",
          "restaurantName": "Entroflex",
          "shortDesc": "Food for foodies",
          "currency": "EUR",
          "deliveryCost": 1.65,
          "rating": 8,
          "minOrder": 14,
          "deliveryTime": "30-40 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/suicide_sue_sitzmoeglichkeiten_top10berlin_800x400px.jpg"
        },
        {
          "id": "6109200e2132d49f1bd63352",
          "restaurantName": "Pyrami",
          "shortDesc": "A lovely place to eat",
          "currency": "EUR",
          "deliveryCost": 1.98,
          "rating": 6,
          "minOrder": 15,
          "deliveryTime": "30-40 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/never_endling_love_story_essen_800x400px.jpg"
        },
        {
          "id": "6109200eb582d1ae7bc3362c",
          "restaurantName": "Electonic",
          "shortDesc": "A lovely place to eat",
          "currency": "USD",
          "deliveryCost": 0.74,
          "rating": 7,
          "minOrder": 13,
          "deliveryTime": "20-25 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/factory_girl_brot_top10_berlin_800x400px.jpg"
        },
        {
          "id": "6109200e3506f225387debb3",
          "restaurantName": "Uni",
          "shortDesc": "Food for foodies",
          "currency": "EUR",
          "deliveryCost": 1.88,
          "rating": 8,
          "minOrder": 10,
          "deliveryTime": "15-20 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/cafe_morena_franzoesisches_fruehstueck_top10_berlin_800x400px.jpg"
        },
        {
          "id": "6109200ef9b9a59c3103ca2c",
          "restaurantName": "Senmao",
          "shortDesc": "A lovely place to eat",
          "currency": "USD",
          "deliveryCost": 0.14,
          "rating": 9,
          "minOrder": 12,
          "deliveryTime": "30-40 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/kaffee_dpa_top10berlin_0.jpg"
        },
        {
          "id": "6109200ed9cf241742fbbfc1",
          "restaurantName": "Aquasure",
          "shortDesc": "A lovely place to eat",
          "currency": "EUR",
          "deliveryCost": 0.94,
          "rating": 6,
          "minOrder": 13,
          "deliveryTime": "20-25 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/zeit_fuer_brot_brot_backen_tino_pohlmann_top10berlin.jpg"
        },
        {
          "id": "6109200ea4e5f4db16e8a223",
          "restaurantName": "Bovis",
          "shortDesc": "Soul for food",
          "currency": "GBP",
          "deliveryCost": 0.1,
          "rating": 8,
          "minOrder": 14,
          "deliveryTime": "20-25 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_szene-restaurants_provocateur-berlin-goldenphoenix_interior_1400x750.jpg"
        },
        {
          "id": "6109200e4e93661d9f2d6aaf",
          "restaurantName": "Cyclonica",
          "shortDesc": "Food for foodies",
          "currency": "USD",
          "deliveryCost": 2.34,
          "rating": 8,
          "minOrder": 14,
          "deliveryTime": "15-20 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_szene-fruehstueck_porridge_haferbrei_dpa.jpg"
        }
      ]
    },
    "indian": {
      "open": [
        {
          "id": "6109200e56b1f459ec67d8a3",
          "restaurantName": "Shepard",
          "shortDesc": "Soul for food",
          "currency": "EUR",
          "deliveryCost": 0.59,
          "rating": 10,
          "minOrder": 14,
          "deliveryTime": "15-20 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_szene-fruehstueck_paulinski-palme001.jpg"
        },
        {
          "id": "6109200e8cb60707931705da",
          "restaurantName": "Exospace",
          "shortDesc": " Food for fun",
          "currency": "GBP",
          "deliveryCost": 2.16,
          "rating": 6,
          "minOrder": 14,
          "deliveryTime": "15-20 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/cabslam-weltrestaurant-food-szene-fruehstueck_top10berlin_1200x650.jpg"
        },
        {
          "id": "6109200ea2e2c8a0f4870ec6",
          "restaurantName": "Zorromop",
          "shortDesc": "Huger Heaven",
          "currency": "USD",
          "deliveryCost": 2.01,
          "rating": 7,
          "minOrder": 13,
          "deliveryTime": "20-25 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_indische-restaurants_india-club-berlin001.jpg"
        },
        {
          "id": "6109200ebec0666dd2fd5f7d",
          "restaurantName": "Strozen",
          "shortDesc": "Huger Heaven",
          "currency": "GBP",
          "deliveryCost": 0.29,
          "rating": 7,
          "minOrder": 15,
          "deliveryTime": "20-25 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_zaika001.jpg"
        },
        {
          "id": "6109200ec4e50073b148ed83",
          "restaurantName": "Kineticut",
          "shortDesc": "Huger Heaven",
          "currency": "USD",
          "deliveryCost": 2.52,
          "rating": 6,
          "minOrder": 14,
          "deliveryTime": "20-25 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_w-der-imbiss003.jpg"
        },
        {
          "id": "6109200e24bf19c2496eac48",
          "restaurantName": "Kozgene",
          "shortDesc": " Food for fun",
          "currency": "GBP",
          "deliveryCost": 2.44,
          "rating": 9,
          "minOrder": 15,
          "deliveryTime": "20-25 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_indische-restaurants_bombay-cafe-buntys.jpg"
        }
      ],
      "close": [
        {
          "id": "6109200e98cc2aae737cc4d0",
          "restaurantName": "Interodeo",
          "shortDesc": "Soul for food",
          "currency": "GBP",
          "deliveryCost": 0.47,
          "rating": 6,
          "minOrder": 14,
          "deliveryTime": "30-40 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/chutnify_dosas_foto_top10berlin_800x400_2.jpg"
        },
        {
          "id": "6109200e989cf634b81a559f",
          "restaurantName": "Exerta",
          "shortDesc": "Food for foodies",
          "currency": "GBP",
          "deliveryCost": 2.74,
          "rating": 9,
          "minOrder": 15,
          "deliveryTime": "30-40 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/khushi_003_top10berlin_800x600px.jpg"
        },
        {
          "id": "6109200e4831013bb2867f41",
          "restaurantName": "Typhonica",
          "shortDesc": "Paradise",
          "currency": "USD",
          "deliveryCost": 0.77,
          "rating": 9,
          "minOrder": 10,
          "deliveryTime": "30-40 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_cafe-tschuesch002.jpg"
        },
        {
          "id": "6109200e6ae9848a74c2c014",
          "restaurantName": "Pushcart",
          "shortDesc": "Food for foodies",
          "currency": "GBP",
          "deliveryCost": 2.46,
          "rating": 6,
          "minOrder": 10,
          "deliveryTime": "20-25 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_chai-wallahs001.jpg"
        },
        {
          "id": "6109200e4c85efa91ccaf557",
          "restaurantName": "Rodeocean",
          "shortDesc": "Paradise",
          "currency": "EUR",
          "deliveryCost": 0.54,
          "rating": 7,
          "minOrder": 10,
          "deliveryTime": "15-20 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/papadam3_top10berlin_800x600px.jpg"
        }
      ]
    },
    "italian": {
      "open": [
        {
          "id": "6109200e47ef991237c48a52",
          "restaurantName": "Cedward",
          "shortDesc": "Food for foodies",
          "currency": "GBP",
          "deliveryCost": 2.55,
          "rating": 6,
          "minOrder": 13,
          "deliveryTime": "20-25 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_chai-ji.jpg"
        },
        {
          "id": "6109200ec0bc597a5be916a6",
          "restaurantName": "Miraclis",
          "shortDesc": " Food for fun",
          "currency": "EUR",
          "deliveryCost": 0.12,
          "rating": 8,
          "minOrder": 15,
          "deliveryTime": "20-25 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/bar-jeder-vernunft-veranstaltungen-berlin.jpg"
        },
        {
          "id": "6109200ef1946852ffb90b18",
          "restaurantName": "Spherix",
          "shortDesc": " Food for fun",
          "currency": "GBP",
          "deliveryCost": 2.15,
          "rating": 10,
          "minOrder": 10,
          "deliveryTime": "30-40 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_erlebnisgastronomie_wilde-matilde_foto_knutschfleckberlin_6.jpg"
        },
        {
          "id": "6109200ed484749da4a1ec1e",
          "restaurantName": "Updat",
          "shortDesc": "Legendary Eat",
          "currency": "GBP",
          "deliveryCost": 0.18,
          "rating": 7,
          "minOrder": 11,
          "deliveryTime": "30-40 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/slider_drachenhaus_innen_betreiber_1.jpg"
        },
        {
          "id": "6109200e0fff849f2f7b9165",
          "restaurantName": "Isbol",
          "shortDesc": "Soul for food",
          "currency": "USD",
          "deliveryCost": 2.79,
          "rating": 6,
          "minOrder": 12,
          "deliveryTime": "15-20 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/schloss_schwante_tafel_c_malena-baum_2_1400x750.jpeg"
        },
        {
          "id": "6109200e32b42eea3d4423b5",
          "restaurantName": "Overfork",
          "shortDesc": "Food for foodies",
          "currency": "EUR",
          "deliveryCost": 2.82,
          "rating": 6,
          "minOrder": 15,
          "deliveryTime": "20-25 min",
          "speciality": "DINNER",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/kriminalmenue_berlin_die-auftakter_4.jpg"
        },
        {
          "id": "6109200e272e77ae2f6d72d3",
          "restaurantName": "Geekology",
          "shortDesc": "Food for foodies",
          "currency": "EUR",
          "deliveryCost": 1.2,
          "rating": 9,
          "minOrder": 15,
          "deliveryTime": "20-25 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/fullsize_moorlake_download_pressekit_01.jpg"
        },
        {
          "id": "6109200e05b34080de9d016e",
          "restaurantName": "Trollery",
          "shortDesc": "Relax your belly",
          "currency": "GBP",
          "deliveryCost": 1.9,
          "rating": 6,
          "minOrder": 11,
          "deliveryTime": "20-25 min",
          "speciality": "BREAKFAST",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/fullsize_tafelrunde_betreiber_1.jpg"
        },
        {
          "id": "6109200e2244eb64b5cfb392",
          "restaurantName": "Aquoavo",
          "shortDesc": "Relax your belly",
          "currency": "USD",
          "deliveryCost": 2.37,
          "rating": 8,
          "minOrder": 11,
          "deliveryTime": "20-25 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_restaurants-fuer-besondere-anlaesse_frischer-fisch_lightfield-studios.jpg"
        },
        {
          "id": "6109200e13037692e908df16",
          "restaurantName": "Organica",
          "shortDesc": "A lovely place to eat",
          "currency": "USD",
          "deliveryCost": 2.07,
          "rating": 6,
          "minOrder": 13,
          "deliveryTime": "30-40 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_restaurants-fuer-besondere-anlaesse_nobelhart-und-schmutzig_caroline-prange001.jpg"
        },
        {
          "id": "6109200e38c792e1b6118aff",
          "restaurantName": "Comtrak",
          "shortDesc": "Legendary Eat",
          "currency": "USD",
          "deliveryCost": 2.49,
          "rating": 6,
          "minOrder": 14,
          "deliveryTime": "15-20 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/austernbank_restaurant_berlin-gendarmenmarkt_interior_02_1200x600.jpg"
        },
        {
          "id": "6109200eb30f359133c0fdac",
          "restaurantName": "Globoil",
          "shortDesc": "Huger Heaven",
          "currency": "EUR",
          "deliveryCost": 2.09,
          "rating": 7,
          "minOrder": 10,
          "deliveryTime": "30-40 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_restaurants-fuer-besondere-anlaesse_tulus-lotrek_vegetarisches-menue002.jpg"
        },
        {
          "id": "6109200eb668aaf2cc9446ed",
          "restaurantName": "Wrapture",
          "shortDesc": "Paradise",
          "currency": "EUR",
          "deliveryCost": 0.69,
          "rating": 6,
          "minOrder": 12,
          "deliveryTime": "30-40 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_restaurants-fuer-besondere-anlaesse_dae-mon001.jpg"
        },
        {
          "id": "6109200e409c13e493764c5a",
          "restaurantName": "Slofast",
          "shortDesc": "Huger Heaven",
          "currency": "EUR",
          "deliveryCost": 1.88,
          "rating": 6,
          "minOrder": 15,
          "deliveryTime": "15-20 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_restaurants-fuer-besondere-anlaesse_nithan-thai.jpg"
        },
        {
          "id": "6109200e3e844d6bbbde52d5",
          "restaurantName": "Norsul",
          "shortDesc": " Food for fun",
          "currency": "EUR",
          "deliveryCost": 0.89,
          "rating": 6,
          "minOrder": 11,
          "deliveryTime": "30-40 min",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_brasserie-colette-tim-raue_nils-hasenau001.jpg"
        }
      ],
      "close": [
        {
          "id": "6109200ee049a252fbba598c",
          "restaurantName": "Proflex",
          "shortDesc": "Huger Heaven",
          "currency": "EUR",
          "deliveryCost": 0.73,
          "rating": 6,
          "minOrder": 14,
          "deliveryTime": "20-25 min",
          "speciality": "LUNCH",
          "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/25_bikini_neni_1_792.jpg"
        }
      ]
    }
  }