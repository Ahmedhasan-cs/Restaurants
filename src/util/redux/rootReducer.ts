import {combineReducers} from 'redux';
import authReducer from './slice/authenticationSlice';
import cuisinesReducer from './slice/cuisineSlice';

const rootReducer = combineReducers({
  auth: authReducer,
  cuisines: cuisinesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
