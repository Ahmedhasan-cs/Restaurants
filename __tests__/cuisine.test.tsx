import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import {
  CuisinesNavigationProps,
  RouteNames,
} from '@navigators/dashboard/DashboardNavigationProps';
import CuisineListScreen from '@features/dashboard/screens/CuisineList/ui/components/CuisineListScreen';
import cuisinesReducer from '@util/redux/slice/cuisineSlice';
import rootReducer from '@util/redux/rootReducer';
import reducer from '@util/redux/slice/cuisineSlice';
import {cuisinesData} from '@util/api/mocks/cuisines';
import createSagaMiddleware from 'redux-saga';

// Mock navigation
const mockNavigate = jest.fn();
const mockRoute = {params: {}};
const mockNavigation: Partial<CuisinesNavigationProps['navigation']> = {
  navigate: mockNavigate,
};

jest.mock('@util/api/apiManager', () => ({
  API_ENDPOINTS: {
    login: '/login',
    cuisines: '/cuisines',
  },
}));

export const createTestStore = (customState = {}) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState: {
      cuisines: {
        cuisines: cuisinesData,
        loading: false,
        error: null,
      },
      auth: {
        isAuthorized: true,
        loading: false,
        error: null,
      },
    },
    middleware: getDefaultMiddleware => getDefaultMiddleware({thunk: false}), // No sagas in tests
  });
};

export const renderWithProviders = (
  ui,
  {store} = {store: createTestStore()},
) => {
  return render(<Provider store={store}>{ui}</Provider>);
};

describe('CuisineListScreen Tests', () => {
  it('should render cuisines from Redux store', () => {
    const {getByText} = renderWithProviders(
      <CuisineListScreen
        navigation={mockNavigation as any}
        route={mockRoute as any}
      />,
    );

    expect(getByText('Italian')).toBeTruthy();
    expect(getByText('Indian')).toBeTruthy();
    expect(getByText('Chinese')).toBeTruthy();

    expect(getByText('19 places')).toBeTruthy();
    expect(getByText('11 places')).toBeTruthy();
    expect(getByText('16 places')).toBeTruthy();
  });

  it('renders CuisineListScreen correctly and matches snapshot', () => {
    const tree = renderWithProviders(
      <CuisineListScreen
        navigation={mockNavigation as any}
        route={mockRoute as any}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot(); // âœ… Snapshot comparison
  });
});
