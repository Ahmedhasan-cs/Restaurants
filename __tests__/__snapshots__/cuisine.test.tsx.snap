// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CuisineListScreen Tests renders CuisineListScreen correctly and matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "flex": 1,
        "paddingTop": 20,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingHorizontal": 16,
          "paddingTop": 40,
        }
      }
      data={
        [
          {
            "count": 19,
            "title": "chinese",
          },
          {
            "count": 11,
            "title": "indian",
          },
          {
            "count": 16,
            "title": "italian",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "aspectRatio": 2.1,
                  "borderRadius": 16,
                  "justifyContent": "center",
                  "marginVertical": 8,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../src/assets/images/chineseImage.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": undefined,
                      "width": "100%",
                    },
                    {
                      "borderRadius": 16,
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "left": 16,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Chinese
              </Text>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "position": "absolute",
                    "right": 16,
                    "top": 16,
                  }
                }
              >
                19
                 places
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "aspectRatio": 2.1,
                  "borderRadius": 16,
                  "justifyContent": "center",
                  "marginVertical": 8,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../src/assets/images/indianImage.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": undefined,
                      "width": "100%",
                    },
                    {
                      "borderRadius": 16,
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "left": 16,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Indian
              </Text>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "position": "absolute",
                    "right": 16,
                    "top": 16,
                  }
                }
              >
                11
                 places
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                {
                  "aspectRatio": 2.1,
                  "borderRadius": 16,
                  "justifyContent": "center",
                  "marginVertical": 8,
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../src/assets/images/italianImage.png",
                  }
                }
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "height": undefined,
                      "width": "100%",
                    },
                    {
                      "borderRadius": 16,
                      "resizeMode": "cover",
                    },
                  ]
                }
              />
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "left": 16,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Italian
              </Text>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "position": "absolute",
                    "right": 16,
                    "top": 16,
                  }
                }
              >
                16
                 places
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
